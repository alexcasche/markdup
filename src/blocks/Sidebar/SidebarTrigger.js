import styled from 'styled-components'
import { tint } from 'polished'

const SidebarTrigger = styled.div`
  display: flex;
  height: 52px;
  align-items: center;
  padding: 0 20px;
  margin-bottom: 2px;
  background-color: ${props => tint(0.85, props.theme.colors.black)};
  color: ${props => props.theme.colors.grey};
  transition: all 0.25s ease-in-out;
  svg:first-child {
    height: 13px;
    width: 20px;
    text-align: center;
    margin-right: 12px;
  }
  span {
    color: inherit;
    flex-grow: 1;
    text-align: left;
    text-transform: uppercase;
    font-size: 14px;
    font-weight: 500;
    letter-spacing: 2px;
  }
  svg:last-child {
    width: 12px;
  }
  &:hover {
    cursor: pointer;
    background-color: ${props => tint(0.8, props.theme.colors.black)};
    color: ${props => props.theme.colors.green};
  }
  &:focus {
    outline: none;
  }
`

export default SidebarTrigger
