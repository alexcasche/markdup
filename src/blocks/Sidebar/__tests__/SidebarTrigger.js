import SidebarTrigger from '../SidebarTrigger'
import { shade } from 'polished'

describe('SidebarTrigger Element', () => {
  it('renders correctly', () => {
    const tree = render(<SidebarTrigger />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
