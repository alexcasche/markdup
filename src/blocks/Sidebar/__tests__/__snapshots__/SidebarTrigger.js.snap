// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarTrigger Element renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 52px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 20px;
  margin-bottom: 2px;
  background-color: #3e4042;
  color: #ABADB3;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
}

.c0 svg:first-child {
  height: 13px;
  width: 20px;
  text-align: center;
  margin-right: 12px;
}

.c0 span {
  color: inherit;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: left;
  text-transform: uppercase;
  font-size: 14px;
  font-weight: 500;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.c0 svg:last-child {
  width: 12px;
}

.c0:hover {
  cursor: pointer;
  background-color: #4a4b4d;
  color: #34F96E;
}

.c0:focus {
  outline: none;
}

<div
  className="c0"
/>
`;
